core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1054,15 +1053,17 @@ class DAGScheduler(
         return
     }
 
+    // Mark the stage as submitted and send an event on the bus. If there are pending tasks, submit
+    // them, otherwise immediately mark the stage as finished, and submit any child stages.
+    stage.latestInfo.submissionTime = Some(clock.getTimeMillis())
```

Comment:
Interesting, had not noticed the change in behavior.
Since behavior of `SparkListenerStageSubmitted` is unfortunately not documented, I agree that perhaps we should not change the semantics here (I am curious if it actually impacts in reality, but good to err on side of caution).